src/vsr/replica.zig

```diff
@@ -711,7 +711,17 @@ pub fn ReplicaType(
                 // that SV to a DVC (dropping the hooks), and never finished the view change.
                 if (op_head == null) {
                     assert(self.view > self.log_view);
-                    op_head = self.journal.op_maximum();
+                    op_head = @max(self.journal.op_maximum(), self.op_checkpoint());
+
+                    if (op_head.? == self.op_checkpoint()) {
+                        const header_checkpoint =
+                            &self.superblock.working.vsr_state.checkpoint.header;
+                        assert(header_checkpoint.op == self.op_checkpoint());
+
+                        if (!self.journal.has(header_checkpoint)) {
+                            self.journal.set_header_as_dirty(header_checkpoint);
+                        }
+                    }
```

Comment:
What about something like:

```suggestion

                    if (self.journal.op_maximum() < self.op_checkpoint()) {
                        const header_checkpoint =
                            &self.superblock.working.vsr_state.checkpoint.header;
                        assert(header_checkpoint.op == self.op_checkpoint());

                        self.journal.set_header_as_dirty(header_checkpoint);
                    }
                    op_head = self.journal.op_maximum();
```

That makes it more clear that this branch's purpose is to ensure the `op_head` doesn't lag behind the `op_checkpoint`.

Reply:
Ah, yes! Nice way to make the intention clearer. 😄